crates/matrix-sdk/src/room/knock_requests.rs

```diff
@@ -176,24 +193,24 @@ mod tests {
 
         let room = server.sync_joined_room(&client, room_id).await;
 
-        let join_request = mock_join_request(&room, None);
+        let knock_request = mock_knock_request(&room, None);
 
         // The /ban endpoint must be called once
         server.mock_ban_user().ok().mock_once().mount().await;
 
-        // When we decline the join request and ban the user from the room
-        join_request
+        // When we decline the knock request and ban the user from the room
+        knock_request
             .decline_and_ban(None)
             .await
             .expect("Failed to decline the request and ban the user");
     }
 
-    fn mock_join_request(room: &Room, event_id: Option<&EventId>) -> JoinRequest {
-        JoinRequest::new(
+    fn mock_knock_request(room: &Room, event_id: Option<&EventId>) -> KnockRequest {
```

Comment:
Can we call this `make_knock_request`, so one is not confused that this is not mocking an endpoint? I've come to read the `mock_` prefix as something for endpoints only…